compose/src/main/java/com/bottlerocketstudios/compose/devoptions/DevOptions.kt

```diff
@@ -0,0 +1,317 @@
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.bottlerocketstudios.compose.R
+
+data class DevOptionsState(
+    val environmentNames: State<List<String>>,
+    val environmentSpinnerPosition: State<Int>,
+    val baseUrl: State<String>,
+    val appVersionName: String,
+    val appVersionCode: String,
+    val appId: String,
+    val buildIdentifier: String,
+    val onEnvironmentChanged: (Int) -> Unit,
+    val onRestartCtaClick: () -> Unit,
+    val onForceCrashCtaClicked: () -> Unit
+)
+
+@Composable
+fun DevOptionsScreen(state: DevOptionsState) {
+    Scaffold(floatingActionButton = { FabLayout(state.onRestartCtaClick) }) {
+        Column(
+            modifier = Modifier.fillMaxSize()
+        ) {
+            DropdownEnvMenu(state)
+            ScrollContent(state)
+        }
+    }
+}
+
+@Composable
+private fun FabLayout(onFabClick: () -> Unit) {
+    FloatingActionButton(
+        onClick = { onFabClick() },
+        backgroundColor = MaterialTheme.colors.secondary,
+        content = {
+            Icon(
+                painter = painterResource(id = R.drawable.ic_restart_24),
+                contentDescription = "Restart",
+                tint = Color.White
+            )
+        }
+    )
+}
+
+@Composable
+private fun ScrollContent(state: DevOptionsState) {
+    Column {
+        LazyColumn(
+            modifier = Modifier
+                .padding(start = 12.dp, end = 12.dp)
+                .fillMaxHeight()
+                .fillMaxWidth()
+                .weight(1F)
+        ) {
+            item {
+                CardDivider(Color.Transparent)
+                CardLayout { EnvironmentCard(baseUrl = state.baseUrl.value) }
+                CardDivider(MaterialTheme.colors.primary)
+            }
+            item {
+                CardLayout { AppInfoCard(state = state) }
+                CardDivider(MaterialTheme.colors.primary)
+            }
+            item {
+                CardLayout { MiscFunctionalityCard(onForceCrashCtaClick = state.onForceCrashCtaClicked) }
+                CardDivider(Color.Transparent)
+            }
+        }
+    }
+}
+
+@Composable
+private fun BottomButton(buttonText: String, onClick: () -> Unit) {
+    Button(
+        onClick = { onClick() },
+        modifier = Modifier
+            .padding(8.dp)
+            .wrapContentHeight()
+            .fillMaxWidth(),
+        colors = ButtonDefaults.textButtonColors(
+            backgroundColor = MaterialTheme.colors.primary
+        ),
+        shape = RoundedCornerShape(7.dp),
```

Comment:
Why 7.dp. ?  Just curious as odd numbers sometimes seem well...... odd. :)

Reply:
This was brought over as is fom what was there which I assume was converted as is from the xml that existed before. That said, its dev options so we can do what we want. I'll be tweaking this during my next PR.

Reply:
Ok, TBH I was asking a leading question....  :)  RoundedCornerShape can also accept an integer between 0 and 100 which represents a percent to round off.  So RoundedCornerShape(50) gives a perfect rounded end without having to compute half the height (which is what I suspected was happening here).